src/types/gecko-profile.js

```diff
@@ -177,41 +177,34 @@ export type GeckoCounter = {|
   |},
 |};
 
-export type GeckoProfileMeta = {|
-  interval: Milliseconds,
+/* This meta object is used in subprocesses profiles.
+ * Using https://searchfox.org/mozilla-central/rev/7556a400affa9eb99e522d2d17c40689fa23a729/tools/profiler/core/platform.cpp#1829
+ * as source of truth. (Please update the link whenever there's a new property).
+ * */
+export type GeckoProfileShortMeta = {|
+  version: number,
   startTime: Milliseconds,
   shutdownTime: Milliseconds | null,
-  abi: string,
-  // The extensions property landed in Firefox 60, and is only optional because
-  // older profile versions may not have it. No upgrader was written for this change.
-  extensions?: GeckoExtensionMeta,
   categories: CategoryList,
-  misc: string,
-  oscpu: string,
-  platform: string,
-  processType: number,
-  product: string,
+|};
+
+/* This meta object is used on the top level profile object.
+ * Using https://searchfox.org/mozilla-central/rev/7556a400affa9eb99e522d2d17c40689fa23a729/tools/profiler/core/platform.cpp#1829
+ * as source of truth. (Please update the link whenever there's a new property).
+ * */
+export type GeckoProfileFullMeta = {|
+  ...GeckoProfileShortMeta,
+  interval: Milliseconds,
   stackwalk: 0 | 1,
   // This value represents a boolean, but for some reason is written out as an int
   // value as the previous field.
   // It's 0 for opt builds, and 1 for debug builds.
   // This property was added to Firefox Profiler a long time after it was added to
   // Firefox, that's why we don't need to make it optional for gecko profiles.
   debug: 0 | 1,
-  toolkit: string,
-  version: number,
-  // The appBuildID, sourceURL, physicalCPUs and logicalCPUs properties landed
-  // in Firefox 62, and are only optional because older processed profile
-  // versions may not have them. No upgrader was written for this change.
-  appBuildID?: string,
-  sourceURL?: string,
-  physicalCPUs?: number,
-  logicalCPUs?: number,
-  // This boolean indicates whether this gecko profile includes already
-  // symbolicated frames. This will be missing for profiles coming from Gecko
-  // (which indicates that they'll need to be symbolicated) but may be specified
-  // for profiles imported from other formats (eg: linux perf).
-  presymbolicated?: boolean,
+  gcpoison: 0 | 1,
```

Comment:
Is it risky to not have these properties as `?` types? It wasn't clear from hg blame when this was introduced.

Reply:
ah, good point, I'll try to find when it's been added and otherwise make them optional.

Reply:
gcpoison is from bug 1326134, landed Dec 28 2016.
asyncstacks is from bug 1326134 as well, landed same date.

so I think it's fine to keep the non-optional after all :)